elyra/pipeline/validation.py

```diff
@@ -165,17 +165,32 @@ def _validate_pipeline_structure(self, pipeline: dict, response: ValidationRespo
         elif 'version' not in primary_pipeline['app_data']:
             response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
                                  message="Pipeline app_data 'version' is missing from primary pipeline")
-        elif not int(primary_pipeline['app_data']['version']) > 0:
+
+        # validate pipeline version compatibility
+        pipeline_version = int(primary_pipeline['app_data']['version'])
```

Comment:
There are several cast operations in the validation code (like this one) that would cause fatal (uncaught) errors, resulting in an invalid validation response that might lead to cascading failures.

Reply:
this will be resolved with the "abstraction layer"